crates/bin/pindexer/src/dex_ex/schema.sql

```diff
@@ -82,3 +82,105 @@ CREATE TABLE IF NOT EXISTS dex_ex_metadata (
   -- The asset id to use for prices in places such as the aggregate summary.
   quote_asset_id BYTEA NOT NULL
 );
+
+CREATE TABLE IF NOT EXISTS dex_ex_position_state (
+  -- Call this rowid to distinguish it from the position ID.
+  rowid SERIAL PRIMARY KEY,
+  -- Immutable position data, defining the trading function.
+  position_id BYTEA NOT NULL UNIQUE,
+  asset_1 BYTEA NOT NULL,
+  asset_2 BYTEA NOT NULL,
+  p NUMERIC(39) NOT NULL,
+  q NUMERIC(39) NOT NULL,
+  close_on_fill BOOLEAN NOT NULL,
+  fee_bps INTEGER NOT NULL,
+  effective_price_1_to_2 FLOAT8 NOT NULL,
+  effective_price_2_to_1 FLOAT8 NOT NULL,
+  position_raw BYTEA NOT NULL,
+  -- The time and height at which the position was opened, and its initial reserves.
+  opening_time TIMESTAMPTZ NOT NULL,
+  opening_height INTEGER NOT NULL,
+  opening_tx BYTEA,
+  opening_reserves_rowid INTEGER NOT NULL,
+  -- The time and height at which the position was closed, if it was closed.
+  closing_time TIMESTAMPTZ,
+  closing_height INTEGER,
+  closing_tx BYTEA
+);
+
+CREATE INDEX ON dex_ex_position_state (position_id);
+CREATE INDEX ON dex_ex_position_state (opening_tx);
+
+CREATE TABLE IF NOT EXISTS dex_ex_position_reserves (
+  rowid SERIAL PRIMARY KEY,
+  position_id BYTEA NOT NULL,
+  height INTEGER NOT NULL,
+  time TIMESTAMPTZ NOT NULL,
+  reserves_1 NUMERIC(39) NOT NULL,
+  reserves_2 NUMERIC(39) NOT NULL
+);
+
+CREATE INDEX ON dex_ex_position_reserves (position_id, height, rowid);
+
+CREATE TABLE IF NOT EXISTS dex_ex_position_executions (
+  rowid SERIAL PRIMARY KEY,
+  position_id BYTEA NOT NULL,
+  height INTEGER NOT NULL,
+  time TIMESTAMPTZ NOT NULL,
+  reserves_rowid INTEGER NOT NULL,
+  -- The input amount of asset 1.
+  delta_1 NUMERIC(39) NOT NULL,
+  -- The input amount of asset 2.
+  delta_2 NUMERIC(39) NOT NULL,
+  -- The output amount of asset 1.
+  lambda_1 NUMERIC(39) NOT NULL,
+  -- The output amount of asset 2.
+  lambda_2 NUMERIC(39) NOT NULL,
+  -- The fee amount paid in asset 1.  
+  fee_1 NUMERIC(39) NOT NULL,
+  -- The fee amount paid in asset 2.
+  fee_2 NUMERIC(39) NOT NULL,
+  -- The context the execution happened in
+  context_asset_start BYTEA NOT NULL,
+  context_asset_end BYTEA NOT NULL
+);
+
+CREATE INDEX ON dex_ex_position_executions (height);
+CREATE INDEX ON dex_ex_position_executions (position_id, height, rowid);
+
+CREATE TABLE IF NOT EXISTS dex_ex_position_withdrawals (
+  rowid SERIAL PRIMARY KEY,
+  position_id BYTEA NOT NULL,
+  height INTEGER NOT NULL,
+  time TIMESTAMPTZ NOT NULL,
```

Comment:
Also redundant if we reference the reserves

Reply:
Same reasoning